deploy/deploy-centralvenv.sh

```diff
@@ -0,0 +1,309 @@
+wmTag="2.0.0"                       # wmcore tag
+serPatch=""                         # a list of service patches to be applied
+serNameToPatch=""                   # a list of service Names to patch
+vmName=""                           # hostname for central services
+vmName=${vmName%%.*}
+pythonCmd=/usr/bin/python3
+
+
+### Searching for the mandatory and optional arguments:
+# export OPTIND=1
+while getopts ":v:t:c:s:p:l:h" opt; do
+    case ${opt} in
+        v)
+            venvPath=$OPTARG
+            venvPath=$(realpath $venvPath)
+            ;;
+        t)
+            wmTag=$OPTARG
+            ;;
+        c)
+            vmName=$OPTARG
+            vmName=${vmName%%.*}
+            ;;
+        s)
+            serNameToPatch=$OPTARG
+            ;;
+        p)
+            serPatch=$OPTARG
+            ;;
+        l)
+            componentList=$OPTARG
+            ;;
+        h)
+            help
+            ;;
+        \? )
+            msg="Invalid Option: -$OPTARG"
+            usage "$msg"
+            ;;
+        : )
+            msg="Invalid Option: -$OPTARG requires an argument"
+            usage "$msg"
+            ;;
+    esac
+done
+
+
+# check for mandatory parameters:
+[[ -z $vmName ]] && usage "Missing mandatory argument: -c <central_services>"
+[[ -z $wmTag ]] && usage "Missing mandatory argument: -t <wmcore_tag"
+
+# setting some more paths
+wmSrcPath=${venvPath}/${wmSrcRelPath}  # WMCore source code target path
+wmDepPath=${venvPath}/sw               # WMCore deployment target path
+wmCfgPath=${venvPath}/config           # WMCore cofig target path
+wmAuthPath=${venvPath}/auth            # WMCore auth target path
+wmTmpPath=${venvPath}/tmp              # WMCore tmp path
+
+startSetupVenv()
+{
+    echo "======================================================="
+    echo "Deployment parameters:"
+    echo "-------------------------------------------------------"
+    echo "componentList: $componentList"
+    echo "venvPath: $venvPath"
+    echo "wmSrcPath: $wmSrcPath"
+    echo "wmDepPath: $wmDepPath"
+    echo "wmCfgPath: $wmCfgPath"
+    echo "wmAuthPath: $wmAuthPath"
+    echo "wmTag: $wmTag"
+    echo "serPatch: $serPatch"
+    echo "serNameToPatch: $serNameToPatch"
+    echo "vmName: $vmName"
+    echo "======================================================="
+    echo -n "Continue? [y]: "
+    read x && [[ $x =~ (n|N) ]] && exit 1
+    echo "..."
+}
+
+initSetupVenv()
+{
+    # Initial setup
+    echo
+    echo "======================================================="
+    echo -n "Assuming this is NOT an initial installation! Is it correct? [y]: "
+    read x && [[ $x =~ (n|N) ]] && {
+        echo "Initial setup ..."
+        echo "Resolving system packages dependencies:"
+        # TODO: Here to probe for the underlying packaging system
+
+        sudo -l
+        sudo yum -y install python3 git zip unzip gcc python-devel mariadb mariadb-devel openssl openssl-devel
```

Comment:
Why `yum`? I thought the script should be independent from Linux package manager? How this will work on debian distros? Do we really need this in a script? I suggest to remove it completely since it add additional function to check if proper set of libraries is present on a system.

Reply:
Moreover, I doubt that this should be enforced either. For instance, to developed or fix some code I may not need mariadb at all. It may only be required when I run specific service. You must identify bare minimum of libraries which are **REQUIRED** to run WMCore services, i.e. in other words without which packages WMCore service will not start. For instance I may not need zip, gcc, python-devel and even openssl to run python.

Reply:
I can remove it. This list of packages here is only the minimal set that I have accumulated throughout the multiple runs on a brand new VM created inside the CERN Openstack, which were breaking the setup. And yes you are correct we are aiming to manage mostly python based dependencies, so those  could be left outside this script and let the one who is about to deploy them to handle the missing pieces in advance.

Reply:
I have added the list of OS and non-python packages/libraries dependencies per WMCore service in the help message of the script. So as we discussed long ago those are not going to be on this script to resolve them. We are going to use some  other means to do so.

@vkuznet  I am Closing this conversation here as well.